### PR TITLE
fix(proxy): change announcement strategy

### LINKED ISSUE #1143: Announcements of projects dont land with current diffing and long period
In the recent party projects only reliably landed on the seed if we reduced the announcement period and always announce everything.

### PR DESCRIPTION
**Pull request by xla**
Addresses project propagations by announcing up on creation.
Additionally lowers the default announce period significantly.

Fixes #1143

### CODE REVIEW

**Comment by FintanH** [comment]
Should probably put this in draft until https://github.com/radicle-dev/radicle-link/pull/401 lands if we're gonna be pointing to its commit :)

File: proxy/coco/src/peer/gossip.rs [hunk 1]
```diff
@@ -8,12 +8,12 @@ use librad::{
...
 
 /// Announce a new rev for the `urn`.
-pub async fn announce(state: &State, urn: &RadUrn, rev: &Oid) {
+pub async fn announce(state: &State, urn: &RadUrn, rev: Option<Oid>) {
     let protocol = state.api.protocol();
     protocol
         .announce(Gossip {
             urn: urn.clone(),
-            rev: Some(Rev::Git((*rev).into())),
+            rev: rev.map(|rev| Rev::Git((rev).into())),
```
Comment (FintanH) [inline proxy/coco/src/peer/gossip.rs:16]: What's the reason for doing this? :thinking:

Comment (xla) [reply proxy/coco/src/peer/gossip.rs:16]: Doing what?

Comment (FintanH) [reply proxy/coco/src/peer/gossip.rs:16]: Right sorry, I mean what's the reason for changing this to always announce the `Rev` to optionally announcing it?

Comment (xla) [reply proxy/coco/src/peer/gossip.rs:16]: We changed the function signature to take an `Option`, so we won't always hav a rev to announce. This was necessary to support the proect creation case where we solely interested in the urn and nothing else.

Comment (FintanH) [reply proxy/coco/src/peer/gossip.rs:16]: Ah makes sense, perfect :ok_hand:

Comment (geigerzaehler) [inline proxy/coco/src/peer/gossip.rs:16]: Looks like there is a redundant pair of parentheses.
```suggestion
            rev: rev.map(|rev| Rev::Git(rev.into())),
```

**Review by geigerzaehler — DISMISSED** [review]

**Review by rudolfs — APPROVED** [review]

**Review by NunoAlexandre — APPROVED** [review]
LG

**Review by cloudhead — APPROVED** [review]

**Review by NunoAlexandre — APPROVED** [review]

**Comment by FintanH** [comment]
This is ready for review once https://github.com/radicle-dev/radicle-link/pull/401 lands :sweat_smile:

**Review by FintanH — APPROVED** [review]
I'm gonna propagate so many commits

**Review by rudolfs — APPROVED** [review]

**Pull request merged**